pkg/sink/initialization_test.go

```diff
@@ -101,3 +111,54 @@ func Test_GetArgs_error(t *testing.T) {
 		})
 	}
 }
+
+func TestWaitForEventlistener(t *testing.T) {
+	eventListenerName := "my-eventlistener"
+	namespace := "my-namespace"
+	eventListener := &triggersv1.EventListener{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      eventListenerName,
+			Namespace: namespace,
+		},
+		Spec: triggersv1.EventListenerSpec{},
+	}
+	ctx, _ := rtesting.SetupFakeContext(t)
+	test.SeedResources(t, ctx, test.Resources{EventListeners: []*triggersv1.EventListener{eventListener}})
+	r := Sink{
+		EventListenerName:      eventListenerName,
+		EventListenerNamespace: namespace,
+		EventListenerLister:    eventlistenerinformer.Get(ctx).Lister(),
+	}
+
+	r.WaitForEventListenerOrDie()
+}
+
+func TestWaitForEventlistener_Fatal(t *testing.T) {
+	if os.Getenv("EL_FATAL_CRASH") == "1" {
```

Comment:
i executed locally this testcase and i see the code under if block never runs until we set env `EL_FATAL_CRASH` 
If thats the case should we keep this? or am i missing something here :thinking:

Reply:
so, the reason that i do this indirection is because the fatal execution will kill the `go test` process. The way we get around this is by creating a subprocess that then fails and adds environment variable that forces the crash.

If the tests were run with EL_FATAL_CRASH in the root process, the `Fatalw` invocation in the backoff would cause the tests to be marked as failed in their entirety